### PR TITLE
tikv-ctl: support hex <-> escaped converting

### PR DESCRIPTION
**Pull request by choleraehyq**
@zhangjinpeng1987 @siddontang PTAL
Signed-off-by: Cholerae Hu <huyingqian@pingcap.com>

### CODE REVIEW

**Comment by siddontang** [comment]
why only mvcc supports hex? I think we should support it for all sub commands.

**Comment by choleraehyq** [comment]
I will provide another sub command to convert from hex to string. Support it for all sub commands is much more complex.

File: src/bin/tikv-ctl.rs [hunk 1]
```diff
@@ -132,6 +142,14 @@ fn main() {
                 .short("e")
                 .takes_value(false)
                 .help("set it when the key is already encoded."))
+            .arg(Arg::with_name("hex")
```
Comment (siddontang) [inline src/bin/tikv-ctl.rs]: I don't think we need these now, only convert command is enough.

File: src/bin/tikv-ctl.rs [hunk 2]
```diff
@@ -141,6 +151,20 @@ fn main() {
                 .help("set commit_ts as filter")));
     let matches = app.clone().get_matches();
 
+    let hex_key = matches.value_of("hex");
```
Comment (siddontang) [inline src/bin/tikv-ctl.rs]: match hex-to-escaped here?

File: src/bin/tikv-ctl.rs [hunk 3]
```diff
@@ -253,6 +277,17 @@ pub struct MvccKv<T> {
     value: T,
 }
 
+fn str_to_hex(key_prefix: &str) -> Vec<u8> {
```
Comment (siddontang) [inline src/bin/tikv-ctl.rs]: seem this is not to_hex, but from hex?

File: src/bin/tikv-ctl.rs (hunk 2 above)
Comment (choleraehyq) [inline src/bin/tikv-ctl.rs]: Sorry for these stupid wrongs...

File: src/bin/tikv-ctl.rs [hunk 4]
```diff
@@ -253,6 +277,17 @@ pub struct MvccKv<T> {
     value: T,
 }
 
+fn hex_str_to_hex_vec(key_prefix: &str) -> Vec<u8> {
```
Comment (siddontang) [inline src/bin/tikv-ctl.rs]: The prefix naming here is not proper, it is a key, not a prefix. 

I prefer using `from_hex(key: &str)` directly.

**Comment by siddontang** [comment]
LGTM
PTAL @zhangjinpeng1987

**Pull request merged**